feat(docker): change the way the request key can be generated (#1117)
bonitasoft/bonita-doc: md/bonita-docker-installation.md

First generate a request key into a container with a specific hostname (-h):

```

> docker run --rm --name=bonita -h bonita -ti quay.io/bonitasoft/bonita-subscription:${varVersion}.0 /bin/bash ./generateRequestKey.sh

Reviewer: Of course :+1:

```

> Answer the questions related to the license you want. This will print out the request key.

Reviewer: ```suggestion
Answer the questions related to the license you want. This will print out the request key and exit automatically from the running container.
```

> Retrieve the licence from the customer portal and place into one folder that will be mounted as volume of the docker image:

Reviewer: ```suggestion
Retrieve the licence from the customer portal and place into one folder that will be mounted as a volume of the docker container. In this example, we will use `~/bonita-lic/.`.
```

```

> cp ~/Downloads/BonitaSubscription-7.7-Cloud_Techuser-bonita-20170124-20170331.lic ~/bonita-lic/.

Reviewer: :question: Why change for ~/bonita-lic/ ?

```

Alternatively, you can create a named persistent volume in docker for keeping license file. See [docker documentation on volumes](https://docs.docker.com/storage/volumes/).

### Start the container

Re-create a new Bonita container with the same hostname (-h) and this host directory mounted (-v) :

```
docker run --name bonita -h bonita -v ~/bonita-lic/:/opt/bonita_lic/ -d -p 8080:8080 quay.io/bonitasoft/bonita-subscription:${varVersion}.0
```

> where `~/bonita-lic/` is the folder containing license file.

Reviewer: This line arrives too late. see comment above
```suggestion
```

::: info

> This will only add the initial license to the Bonita Runtime. To update a license on an existing installation see [Update configuration and license section](#section-update-configuration)

Reviewer: ```suggestion
This will only add the initial license to the Bonita Runtime. To renew a license on an existing installation see [Update configuration and license section](#section-update-configuration)
```

* Stop and destroy the running Bonita container.
* Play the migration script on your Bonita database see [migrate the platform from an earlier version of Bonita](migrate-from-an-earlier-version-of-bonita-bpm.md#migrate).
* Get the new Bonita docker image, as explained above.
* Update the license, see [Update configuration and license section](#section-update-configuration)
* Start a new Bonita container.

<a id="section-update-configuration" />

## Update configuration and license

> The license file and configuration files are updated using the Setup tool.

Reviewer: ```suggestion
Once renewed from Bonita Customer Portal, the license file and the configuration files are updated using the Setup tool.
```

Setup tool can be used outside of the Docker container directly by downloading the Tomcat bundle and running it from there.

::: info

> The setup tool need to be able to access the database. Because of that, if the database is in a docker container, its port must be exposed to the host.

Reviewer: ```suggestion
The setup tool needs to be able to access the database. Because of that, if the database is in a docker container, its port must be exposed to the host.
```

:::

> See [setup tool page]([setup tool](BonitaBPM_platform_setup.md#update_platform_conf)) for more informations.

Reviewer: Always singular:
```suggestion
See [setup tool page]([setup tool](BonitaBPM_platform_setup.md#update_platform_conf)) for more information.
```
